src/AgDatabase.cs

```diff
@@ -184,6 +184,16 @@ private void AddNewSqlLogin(LoginProperties login)
       Parallel.ForEach(_listener.Secondaries, server => server.AddLogin(login));
     }
 
+    public void AddUser(UserProperties user)
+    {
+      _listener.Primary.AddUser(user, Name);
+    }
+
+    public void DropUser(UserProperties user)
+    {
+      _listener.Primary.DropUser(user, Name);
+    }
```

Comment:
Shouldn't we propagate this to all servers, not just primary?

Reply:
Users are a database level property which synchronize through the AG
https://www.sqlshack.com/synchronize-logins-between-availability-replicas-in-sql-server-always-on-availability-group/

Reply:
This was designed to be called once the database has already joined the availability group, at which point the database-level object will  be replicated on the secondary's. I haven't tested an implementation where the database is not joined to the ag, but I assume in that case propagating this to all servers would be appropriate.

The current implementation could be extended to handle this case as well. It would need to detect whether or not the database is an availability database.